crates/nu-command/src/filesystem/uutil-notes.md

```diff
@@ -0,0 +1,120 @@
+# Using uu_cratename in nushell
+
+## Argument Passing
+
+We need to get arguments form nushell and pass arguments to uumain. I seriously doubt that the nushell argument structures will be 100% compatible with clap, which is what is used in uutils, so there could be some rough edges.
+
+This is a prototype for copy in PR #9463
+```rust
+fn run(
+    &self,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    _input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    // Get the App
+    // let mut app = uu_cp::uu_app();
+    // app.print_help()?;
+
+    // Create uucore::Args somehow from nushell args
+    let s1 = "cp".to_string();
+    let s2 = "-h".to_string();
+    let args = vec![OsString::from(s1), OsString::from(s2)];
```

Comment:
Makes me wonder if we instead want to create a `From/Into` for clap, so we can take the args and convert them using a converter that all the uu commands share.

Reply:
Seems reasonable. We need to construct our arguments so that they're compatible `uucore::Args`. I think this is the trait definition for that. https://github.com/uutils/coreutils/blob/17830ad6f84583a3d552b360dece0c74d7efc2e3/src/uucore/src/lib/lib.rs#L155-L167